### DISCUSSION TITLE
_refln.crystal_id and wavelength_id

### DISCUSSION THREAD

**Discussion by keitaroyam** [project-gemmi/gemmi#155] [About wwPDB depositions]
My colleague who deposited an unmerged cif generated in mxdepo.html (#130) was told to re-upload the cif file by the PDB annotator, because following items were missing:
_refln.crystal_id 
_refln.wavelength_id 

Do you have a similar experience? Could you add these if they are actually needed?

**Comment by wojdyr** [comment]
So far I haven't heard about any objections from annotators.
A few entries from mxdepo.html have been released during the last months and they don't have these items:
7MM1 7ONK 7ONN 7ONO 7ONW 7ONX 7ONY 7ONZ 7RIN

Crystal_id and wavelength_id make sense only for unmerged reflections. Due to software limitations in almost all entries they are included also in merged data. The data table usually starts with:
_refln.crystal_id
_refln.wavelength_id
_refln.scale_group_code
and all three are set dummy values ( `1 1 1`).

Formally, they are not mandatory:
https://mmcif.wwpdb.org/dictionaries/mmcif_pdbx_v50.dic/Categories/refln.html
and the software limitations that made them necessary were partly lifted in March, when the PDB announced [improved support for extended PDBx/mmCIF structure factor files][1].

Could your colleague reply to the PDB annotator that these items are not required? If more than one crystal or wavelength is used, the merged intensity is a weighted average of intensities from all crystals and wavelengths, so it's not possible to meaningfully assign these IDs.

If there are technical problems with processing such files by the OneDep pipeline let me know, I'll try to sort it out. Both mxdepo.html and OneDep are updated from time to time, so it could happen that what used to work previously doesn't work now.

[1]: https://www.wwpdb.org/news/news?year=2021#60638da1931d5660393084c3

**Reply by keitaroyam** [reply]
Thanks for your reply. That totally makes sense. We are waiting for response from the annotator.

**Comment by keitaroyam** [comment]
The annotator was concerned about the warning message on their system. Today the entry was released, and it seems the annotator finally put dummy values for _refln.crystal_id and _refln.wavelength_id.

I hope the system will be clearer so that unnecessary warnings are not shown to annotators.
